File: src/main/java/SymmetricTree/GoKu-gaga/SymmetricTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
  function isMirror (left, right) {
      if (left === null && right === null) {
          return true
      }
      if (left !== null && right !== null && left.val === right.val) {
          return isMirror(left.left, right.right) && isMirror(left.right, right.left)
      }
      return false
  }
  
  return root === null ? true : isMirror(root, root)
};


/** 
 * 执行结果
 * 执行用时 : 88 ms, 在所有JavaScript提交中击败了92.04%的用户
 * 内存消耗 : 35.4 MB, 在所有JavaScript提交中击败了56.59%的用户
*/

File: src/main/java/SymmetricTree/jngdg/SymmetricTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        return isMirror(root,root);
    }
    bool isMirror(TreeNode* t1, TreeNode* t2)
    {
        if(t1 == NULL && t2 == NULL)
            return true;
        if(t1 == NULL || t2 == NULL)
            return false;
        if(t1->val == t2->val)
            return isMirror(t1->left, t2->right) && isMirror(t1->right, t2->left);
        else
            return false;
    }
};